Services/Mail/classes/class.ilMailOptions.php

```diff
@@ -134,15 +139,15 @@ protected function read($firstMail = '', $secondMail = '')
 		$this->incoming_type        = $row->incoming_type;
 		$this->mail_address_option  = (int)$row->mail_address_option >= 3 ? $row->mail_address_option : self::FIRST_EMAIL;
 
-		if ($firstMail === '') {
-			$firstMail  = ilObjUser::_lookupEmail($this->user_id);
+		if ($firstMailAddress === '') {
```

Comment:
Instead of using static methods here, we could use an INNER JOIN in the query above and select both the email and the second email. This reduces the amount of SQL queries when instantiating ilMailOptions.